reference-implementation/to-upstream-wpts/transform-stream/flush.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+promise_test(() => {
+  let flushCalled = false;
+  const ts = new TransformStream({
+    transform() { },
+    flush() {
+      flushCalled = true;
+    }
+  });
+
+  return ts.writable.getWriter().close().then(() => {
+    return assert_true(flushCalled, 'closing the writable triggers the transform flush immediately');
+  });
+}, 'TransformStream flush is called immediately when the writable is closed, if no writes are queued');
+
+// TODO
+test(() => {
+  let flushCalled = false;
+  const ts = new TransformStream({
+    transform() {
+      return new Promise(resolve => setTimeout(resolve, 10));
+    },
+    flush() {
+      flushCalled = true;
+      return new Promise(); // never resolves
+    }
+  });
+
+  const writer = ts.writable.getWriter();
+  writer.write('a');
+  writer.close();
+  assert_false(flushCalled, 'closing the writable does not immediately call flush if writes are not finished');
+
+  let rsClosed = false;
+  ts.readable.getReader().closed.then(() => {
+    rsClosed = true;
+  });
+
+  setTimeout(() => {
+    assert_true(flushCalled, 'flush is eventually called');
+    assert_equals(rsClosed, false, 'if flushPromise does not resolve, the readable does not become closed');
+  }, 50);
+}, 'TransformStream flush is called after all queued writes finish, once the writable is closed');
+
+promise_test(() => {
+  let c;
+  const ts = new TransformStream({
+    start(controller) {
+      c = controller;
+    },
+    transform() {
+    },
+    flush() {
+      c.enqueue('x');
+      c.enqueue('y');
+    }
+  });
+
+  const reader = ts.readable.getReader();
+
+  const writer = ts.writable.getWriter();
+  writer.write('a');
+  writer.close();
+  reader.read().then(result1 => {
```

Comment:
return the promise here, and remove the `.catch(e => t.error(e))` (`t` is not defined)